### PR TITLE
Schema and code changes to get contracts tests passing for all three …

### PR DESCRIPTION
**Pull request by chrigilm**
…sub projects

*Description of changes:*
Added ARN patterns and other needed schema changes
Updated code to behave as contract tests wanted.  
Changes include: 

- Returning FAILURE for Delete and Update when the resource does not exist
- Returning no model upon successful deletion
- Adding missing elements to returned models

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by DJ-C** [comment]
Let's remove t.corp links as they are internal to amazon.

File: aws-globalaccelerator-accelerator/aws-globalaccelerator-accelerator.json [hunk 1]
```diff
@@ -66,7 +68,8 @@
     },
     "AcceleratorArn": {
       "description": "The Amazon Resource Name (ARN) of the accelerator.",
-      "type": "string"
+      "type": "string",
+      "pattern": "^arn:(aws|aws-cn|aws-us-gov|aws-iso|aws-iso-b):(\\w+)::(\\d{12}):(accelerator)/([0-9a-f-]+)$"
```
Comment (DJ-C) [inline aws-globalaccelerator-accelerator/aws-globalaccelerator-accelerator.json:72]: If we change the pattern in future APIs, we will have to make a change here as well. Is it a requirement to have regex here?

File: aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/DeleteHandler.kt [hunk 2]
```diff
@@ -23,7 +24,7 @@ class DeleteHandler : BaseHandler<CallbackContext?>() {
         val inferredCallbackContext = callbackContext ?: CallbackContext(HandlerCommons.NUMBER_OF_STATE_POLL_RETRIES)
         val model = request.desiredResourceState
         val foundAccelerator = HandlerCommons.getAccelerator(model.acceleratorArn, proxy, agaClient, logger)
-                ?: return ProgressEvent.defaultSuccessHandler(model)
+                ?: return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.NotFound, "Accelerator Not Found")
```
Comment (DJ-C) [inline aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/DeleteHandler.kt:27]: Can you confirm if it matches the other customer facing messages?

File: aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/UpdateHandler.kt [hunk 3]
```diff
@@ -143,7 +143,11 @@ class UpdateHandler : BaseHandler<CallbackContext?>() {
     }
 
     private fun byoipIPsUpdated(currentModel: ResourceModel, previousModel: ResourceModel): Boolean {
-        return currentModel.ipAddresses != previousModel.ipAddresses
+        var ipAddressesUpdated = false
+        if  (currentModel.ipAddresses != null && currentModel.ipAddresses != previousModel.ipAddresses) {
```
Comment (DJ-C) [inline aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/UpdateHandler.kt:147]: nit: 2 spaces after 'if'.

File: aws-globalaccelerator-endpointgroup/src/main/kotlin/software/amazon/globalaccelerator/endpointgroup/ReadHandler.kt [hunk 4]
```diff
@@ -43,6 +41,7 @@ class ReadHandler : BaseHandler<CallbackContext>() {
                 this.endpointGroupRegion = endpointGroup.endpointGroupRegion
                 this.endpointConfigurations = getEndpointConfigurations(endpointGroup.endpointDescriptions)
                 this.portOverrides = getPortOverrides(endpointGroup.portOverrides)
+                this.listenerArn =currentModel.listenerArn
```
Comment (DJ-C) [inline aws-globalaccelerator-endpointgroup/src/main/kotlin/software/amazon/globalaccelerator/endpointgroup/ReadHandler.kt:44]: nit: space after =

File: aws-globalaccelerator-endpointgroup/src/main/kotlin/software/amazon/globalaccelerator/endpointgroup/UpdateHandler.kt [hunk 5]
```diff
@@ -24,13 +24,10 @@ class UpdateHandler : BaseHandler<CallbackContext>() {
...
                 ?: CallbackContext(stabilizationRetriesRemaining = HandlerCommons.NUMBER_OF_STATE_POLL_RETRIES, pendingStabilization = false)
         val model = request.desiredResourceState
-        HandlerCommons.getEndpointGroup(model.endpointGroupArn, proxy, agaClient, logger)
-                ?: return ProgressEvent.defaultFailureHandler(
-                        Exception("Failed to find endpoint group with arn:[${model.endpointGroupArn}]"),
-                        HandlerErrorCode.NotFound
-                )
         return if (!inferredCallbackContext.pendingStabilization) {
-            updateEndpointGroup(model, request.previousResourceState, proxy, agaClient, logger)
+            HandlerCommons.getEndpointGroup(model.endpointGroupArn, proxy, agaClient, logger)
```
Comment (DJ-C) [inline aws-globalaccelerator-endpointgroup/src/main/kotlin/software/amazon/globalaccelerator/endpointgroup/UpdateHandler.kt:28]: Can you please confirm if it matches other customer facing messages?

File: aws-globalaccelerator-endpointgroup/src/test/kotlin/software/amazon/globalaccelerator/endpointgroup/ReadHandlerTest.kt [hunk 6]
```diff
@@ -102,6 +101,7 @@ class ReadHandlerTest {
         Assertions.assertEquals(response.resourceModel.portOverrides.size, 1)
         Assertions.assertEquals(response.resourceModel.portOverrides[0].listenerPort, portOverrides[0].listenerPort)
         Assertions.assertEquals(response.resourceModel.portOverrides[0].endpointPort, portOverrides[0].endpointPort)
+        Assertions.assertEquals(response.resourceModel.listenerArn, "arn:aws:globalaccelerator::474880776455:accelerator/abcd1234/listener/12341234")
```
Comment (DJ-C) [inline aws-globalaccelerator-endpointgroup/src/test/kotlin/software/amazon/globalaccelerator/endpointgroup/ReadHandlerTest.kt:104]: Can we use const for arn?

**Comment by chrigilm** [comment]
> Let's remove t.corp links as they are internal to amazon.

Done

**Review by DJ-C — CHANGES_REQUESTED** [review]
I have left a few comments mainly around security and consistency.

File: aws-globalaccelerator-accelerator/aws-globalaccelerator-accelerator.json (hunk 1 above)
Comment (chrigilm) [reply aws-globalaccelerator-accelerator/aws-globalaccelerator-accelerator.json:72]: There was no requirement. We only had the pattern in one place (EPG arn) and that was not valid according to the validation.  I updated to be valid and added pattern to all other arns.  I can remove if you feel strongly about it.

File: aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/DeleteHandler.kt (hunk 2 above)
Comment (chrigilm) [reply aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/DeleteHandler.kt:27]: Yes.  This was patterned after existing messages in the code.

File: aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/UpdateHandler.kt (hunk 3 above)
Comment (chrigilm) [reply aws-globalaccelerator-accelerator/src/main/kotlin/software/amazon/globalaccelerator/accelerator/UpdateHandler.kt:147]: Fixed

File: aws-globalaccelerator-endpointgroup/src/main/kotlin/software/amazon/globalaccelerator/endpointgroup/UpdateHandler.kt (hunk 5 above)
Comment (chrigilm) [reply aws-globalaccelerator-endpointgroup/src/main/kotlin/software/amazon/globalaccelerator/endpointgroup/UpdateHandler.kt:28]: Yes, patterned after other message in the code.

File: aws-globalaccelerator-accelerator/aws-globalaccelerator-accelerator.json (hunk 1 above)
Comment (DJ-C) [reply aws-globalaccelerator-accelerator/aws-globalaccelerator-accelerator.json]: Discussed and closed offline. We will remove this check.

**Review by DJ-C — APPROVED** [review]

**Comment by DJ-C** [comment]
Schema & handler changes for contract tests.

**Pull request merged**